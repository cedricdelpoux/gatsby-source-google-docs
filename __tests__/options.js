const documentLinks = require("./documents/links.json")
const documentTexts = require("./documents/texts.json")
const {GoogleDocument} = require("../utils/google-document")

test(`"DemoteHeading" option`, () => {
  const googleDocument = new GoogleDocument(
    documentTexts,
    {},
    {demoteHeadings: true}
  )
  expect(googleDocument.toMarkdown()).toMatchSnapshot()
})

test(`Crosslinks between documents`, () => {
  const options = {
    crosslinksPaths: {
      [documentLinks.documentId]: "/relative-path",
      ["unknow"]: "/404",
    },
  }
  const googleDocument = new GoogleDocument(documentLinks, {}, options)
  const documentObject = googleDocument.toMarkdown()
  expect(documentObject).toMatchSnapshot()
})
