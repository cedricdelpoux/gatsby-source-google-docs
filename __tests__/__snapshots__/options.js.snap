// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"DemoteHeading" option 1`] = `
"---
cover: null
---
## Title level 1

### Title level 2

#### Title level 3

##### Title level 4

###### Title level 5

###### Title level 6


**bold**


_italic_


<ins>underline</ins>


~~strikethrough~~ 


<sup>superscript</sup>


<sub>subscript</sub>


**_boldItalic_**


text with **space bold after** and _space italic before_.


		indented text


[Link](https://github.com/cedricdelpoux/gatsby-source-google-docs)


<span style=\\"font-size:1.45em\\">Text bigger</span>


Text Colored


Text with background color
"
`;

exports[`Crosslinks between documents 1`] = `
"---
cover: null
---

[to self](/relative-path)


[to self with user id](/relative-path)


[to self with edit](/relative-path)


[to self with preview](/relative-path)


[unknown url](https://docs.google.com/document/d/unknown)
"
`;
